lib/index.js

```diff
@@ -74,6 +79,21 @@ module.exports = function(playlistURL, options) {
       totalItems++;
       requestQueue.push(item, onQueuedEnd);
     });
+    parser.on('ytdl-end', function() {
+      fetchingPlaylist = false;
+      delete parser;
+      refreshPlaylist();
+      rpts = 0;
+      ytft = false;
+    });
+    parser.on('ytdl-error', function() {
```

Comment:
When the parser has an internal error when parsing time or something like that, this will be called. If rpts (see above) is more than or equal to 5, then error out and stop with 'Too Many Requests' signaling an internal error. Optionally, we might want to add a second argument called "err" as when this event is called it does come with reasoning.